src/libsyntax/ext/expand.rs

```diff
@@ -454,7 +454,7 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
         };
         let path = &mac.node.path;
 
-        let ident = ident.unwrap_or(keywords::Invalid.ident());
+        let ident = ident.unwrap_or_else(|| keywords::Invalid.ident());
```

Comment:
@petrochenkov I don't think clippy is able to determine whether a function is heavy or not. It recommends `unwrap_or_else` whenever the inner expression is not a variable/primitive.